<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">

<!--Converted with jLaTeX2HTML 2002-2-1 (1.70) JA patch-2.0
patched version by:  Kenshi Muto, Debian Project.
* modified by:  Shige TAKENO
LaTeX2HTML 2002-2-1 (1.70),
original version by:  Nikos Drakos, CBLU, University of Leeds
* revised and updated by:  Marcus Hennecke, Ross Moore, Herb Swan
* with significant contributions from:
  Jens Lippmann, Marek Rouchal, Martin Wilck and others -->
<HTML>
<HEAD>
<TITLE>alignment.salign() -- align two or more sequences/structures of proteins</TITLE>
<META NAME="description" CONTENT="alignment.salign() -- align two or more sequences/structures of proteins">
<META NAME="keywords" CONTENT="manual">
<META NAME="resource-type" CONTENT="document">
<META NAME="distribution" CONTENT="global">

<META NAME="Generator" CONTENT="jLaTeX2HTML v2002-2-1 JA patch-2.0">
<META HTTP-EQUIV="Content-Style-Type" CONTENT="text/css">

<LINK REL="STYLESHEET" HREF="manual.css">

<LINK REL="next" HREF="node85.html">
<LINK REL="previous" HREF="node83.html">
<LINK REL="up" HREF="node62.html">
<LINK REL="next" HREF="node85.html">
</HEAD>

<BODY >
<!--Navigation Panel-->
<A NAME="tex2html1803"
  HREF="node85.html">
<IMG WIDTH="37" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="next" SRC="next.png"></A> 
<A NAME="tex2html1797"
  HREF="node62.html">
<IMG WIDTH="26" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="up" SRC="up.png"></A> 
<A NAME="tex2html1791"
  HREF="node83.html">
<IMG WIDTH="63" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="previous" SRC="prev.png"></A> 
<A NAME="tex2html1799"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="contents" SRC="contents.png"></A> 
<A NAME="tex2html1801"
  HREF="node152.html">
<IMG WIDTH="43" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="index" SRC="index.png"></A> 
<BR>
<B> Next:</B> <A NAME="tex2html1804"
  HREF="node85.html">alignment.to_profile()   convert</A>
<B> Up:</B> <A NAME="tex2html1798"
  HREF="node62.html">The alignment class: comparison</A>
<B> Previous:</B> <A NAME="tex2html1792"
  HREF="node83.html">alignment.malign3d()   align</A>
 &nbsp; <B>  <A NAME="tex2html1800"
  HREF="node1.html">Contents</A></B> 
 &nbsp; <B>  <A NAME="tex2html1802"
  HREF="node152.html">Index</A></B> 
<BR>
<BR>
<!--End of Navigation Panel-->
<!--Table of Child-Links-->
<A NAME="CHILD_LINKS"><STRONG>Subsections</STRONG></A>

<UL>
<LI><A NAME="tex2html1805"
  HREF="node84.html#SECTION007422100000000000000">Multiple alignments of protein structures (and sequences)</A>
</UL>
<!--End of Table of Child-Links-->
<HR>

<H2><A NAME="SECTION007422000000000000000">
alignment.salign() -- align two or more sequences/structures of proteins</A>
</H2> <A NAME="18940"></A>

<P>
<TABLE CELLPADDING=3>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>align3d_trf</I><A NAME="19202"></A> = </FONT><FONT SIZE="-1"><TT>&lt;logical:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>False</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">whether to transform the distances before dynamic programming</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>align_block</I><A NAME="19214"></A> = </FONT><FONT SIZE="-1"><TT>&lt;integer:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>0</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">the last sequence in the first block of sequences</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>alignment_type</I><A NAME="19226"></A> = </FONT><FONT SIZE="-1"><TT>&lt;string:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>'PROGRESSIVE'</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">'PAIRWISE' 'TREE' 'PROGRESSIVE' for SALIGN</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>align_what</I><A NAME="19238"></A> = </FONT><FONT SIZE="-1"><TT>&lt;string:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>'BLOCK'</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">what to align in ALIGN; </FONT><FONT SIZE="-1"><TT>'BLOCK'</TT> </FONT><FONT SIZE="-1"><TT>|</TT> </FONT><FONT SIZE="-1"><TT>'ALIGNMENT'</TT> </FONT><FONT SIZE="-1"><TT>|</TT> </FONT><FONT SIZE="-1"><TT>'LAST'</TT> </FONT><FONT SIZE="-1"><TT>|</TT> </FONT><FONT SIZE="-1"><TT>'PROFILE'</TT></FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>atom_files_directory</I><A NAME="19261"></A> = </FONT><FONT SIZE="-1"><TT>&lt;string:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>'./'</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">input atom files directory list (e.g., </FONT><FONT SIZE="-1"><TT>'dir1:dir2:dir3:./:/'</TT>)</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>comparison_type</I><A NAME="19275"></A> = </FONT><FONT SIZE="-1"><TT>&lt;string:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>'MAT'</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">'MAT' or 'PSSM' for comparing matrices or PSSMs when profiles are compared</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>current_directory</I><A NAME="19287"></A> = </FONT><FONT SIZE="-1"><TT>&lt;logical:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>True</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">whether to write output .fit files to current directory</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>dendrogram_file</I><A NAME="19299"></A> = </FONT><FONT SIZE="-1"><TT>&lt;string:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>''</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">File into which the SALIGN dendrogram is written out</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>ext_tree</I><A NAME="19311"></A> = </FONT><FONT SIZE="-1"><TT>&lt;logical:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>False</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">To read a pre computed tree for SALIGN</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>feature_weights</I><A NAME="19323"></A> = </FONT><FONT SIZE="-1"><TT>&lt;real:6&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>1 0 0 0 0 0</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">feature weights for SALIGN</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>fit_atoms</I><A NAME="19335"></A> = </FONT><FONT SIZE="-1"><TT>&lt;string:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>'CA'</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">atom type(s) being superposed</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>fit_on_first</I><A NAME="19347"></A> = </FONT><FONT SIZE="-1"><TT>&lt;logical:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>False</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">whether or not to optimize alignment in SALIGN</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>fit_pdbnam</I><A NAME="19359"></A> = </FONT><FONT SIZE="-1"><TT>&lt;logical:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>True</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">whether or not to add _fit to the PDB file name in output alifile by SALIGN</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>fit</I><A NAME="19371"></A> = </FONT><FONT SIZE="-1"><TT>&lt;logical:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>True</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">whether to do pairwise least-squares fitting or ALIGN2D alignment</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>fix_offsets</I><A NAME="19383"></A> = </FONT><FONT SIZE="-1"><TT>&lt;real:5&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>0 1000 2000 3000 4000</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">offsets of the ALIGN2D alignment score for "fixed" positions indicated by ' 123456789' in line '_fix_pos'</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>gap_function</I><A NAME="19395"></A> = </FONT><FONT SIZE="-1"><TT>&lt;logical:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>False</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">whether or not to switch on functional gap penalty in salign</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>gap_gap_score</I><A NAME="19407"></A> = </FONT><FONT SIZE="-1"><TT>&lt;real:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>0</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">dissimilarity score for aligning gap with gap, in SALIGN</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>gap_penalties_1d</I><A NAME="19419"></A> = </FONT><FONT SIZE="-1"><TT>&lt;real:2&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>900 50</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">gap creation and extension penalties for sequence/sequence alignment</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>gap_penalties_2d</I><A NAME="19431"></A> = </FONT><FONT SIZE="-1"><TT>&lt;real:9&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>0.35 1.2 0.9 1.2 0.6 8.6 1.2 0 0</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">gap penalties for sequence/structure alignment: helix, beta, accessibility, straightness, and CA-CA distance factor, dst min, dst power, t, structure_profile ; best U,V=-450,0</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>gap_penalties_3d</I><A NAME="19443"></A> = </FONT><FONT SIZE="-1"><TT>&lt;real:2&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>0.0 1.75</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">gap creation and extension penalties for structure/structure superposition</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>gap_residue_score</I><A NAME="19455"></A> = type1 </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>0</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">dissimilarity score for aligning gap with residue, in SALIGN</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>hetatm_io</I><A NAME="19465"></A> = </FONT><FONT SIZE="-1"><TT>&lt;logical:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>False</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">whether to read HETATM coordinates</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>hydrogen_io</I><A NAME="19477"></A> = </FONT><FONT SIZE="-1"><TT>&lt;logical:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>False</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">whether to read hydrogen coordinates</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>improve_alignment</I><A NAME="19489"></A> = </FONT><FONT SIZE="-1"><TT>&lt;logical:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>True</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">whether or not to optimize alignment in SALIGN</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>input_weights_file</I><A NAME="19501"></A> = </FONT><FONT SIZE="-1"><TT>&lt;string:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>''</TT></FONT></TD>
<TD></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>local_alignment</I><A NAME="19513"></A> = </FONT><FONT SIZE="-1"><TT>&lt;logical:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>False</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">whether to do local as opposed to global alignment</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>matrix_comparison</I><A NAME="19525"></A> = </FONT><FONT SIZE="-1"><TT>&lt;string:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>'CC'</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">'CC', 'MAX', 'AVE', - kinds of matrix comparisons</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>matrix_offset</I><A NAME="19537"></A> = </FONT><FONT SIZE="-1"><TT>&lt;real:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>0.00</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">substitution matrix offset for local alignment</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>max_gap_length</I><A NAME="19549"></A> = </FONT><FONT SIZE="-1"><TT>&lt;integer:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>999999</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">maximal length of gap in protein comparisons</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>maxres</I><A NAME="19561"></A> = </FONT><FONT SIZE="-1"><TT>&lt;integer:0&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>0</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">user specified maximal number of residues</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>normalize_pp_scores</I><A NAME="19573"></A> = </FONT><FONT SIZE="-1"><TT>&lt;logical:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>False</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">whether or not to normalize position-position scores in SALIGN</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>no_ter</I><A NAME="19585"></A> = </FONT><FONT SIZE="-1"><TT>&lt;logical:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>False</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">whether to not write TER into PDB</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>n_subopt</I><A NAME="19597"></A> = </FONT><FONT SIZE="-1"><TT>&lt;integer:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>1</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">number of optimal and suboptimal alignments ALIGN/ALIGN2D</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>off_diagonal</I><A NAME="19609"></A> = </FONT><FONT SIZE="-1"><TT>&lt;integer:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>100</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">to speed up the alignment</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>output_weights_file</I><A NAME="19621"></A> = </FONT><FONT SIZE="-1"><TT>&lt;string:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>''</TT></FONT></TD>
<TD></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>output</I><A NAME="19633"></A> = </FONT><FONT SIZE="-1"><TT>&lt;string:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>'LONG'</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">what and/or how to output</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>overhang</I><A NAME="19645"></A> = </FONT><FONT SIZE="-1"><TT>&lt;integer:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>0</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">un-penalized overhangs in protein comparisons</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>rms_cutoffs</I><A NAME="19657"></A> = </FONT><FONT SIZE="-1"><TT>&lt;real:11&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>3.5 3.5 60 60 15 60 60 60 60 60 60</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">cutoffs for RMS, DRMS, Alpha Phi Psi Omega chi1 chi2 chi3 chi4 chi5</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>rr_file</I><A NAME="19669"></A> = </FONT><FONT SIZE="-1"><TT>&lt;string:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>'$(LIB)/as1.sim.mat'</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">input residue-residue scoring file</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>similarity_flag</I><A NAME="19681"></A> = </FONT><FONT SIZE="-1"><TT>&lt;logical:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>False</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">when turned on, the SALIGN command does not convert numbers into a distance sense.</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>smooth_prof_weight</I><A NAME="19693"></A> = </FONT><FONT SIZE="-1"><TT>&lt;real:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>10</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">for smoothing the profile aa frequency with a prior</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>stop_on_error</I><A NAME="19705"></A> = </FONT><FONT SIZE="-1"><TT>&lt;integer:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>1</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">whether to stop on error</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>subopt_offset</I><A NAME="19717"></A> = </FONT><FONT SIZE="-1"><TT>&lt;real:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>0.0</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">offset for residue-residue score in getting suboptimals in ALIGN/ALIGN2D</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>substitution</I><A NAME="19729"></A> = </FONT><FONT SIZE="-1"><TT>&lt;logical:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>False</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">whether to use the background in PSSM comparison</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>water_io</I><A NAME="19741"></A> = </FONT><FONT SIZE="-1"><TT>&lt;logical:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>False</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">whether to read water coordinates</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>weights_type</I><A NAME="19753"></A> = </FONT><FONT SIZE="-1"><TT>&lt;string:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>'SIMILAR'</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">or 'DISTANCE' -&gt; for the kind of substitution values</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>write_fit</I><A NAME="19765"></A> = </FONT><FONT SIZE="-1"><TT>&lt;logical:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>False</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">whether to write out fitted coordinates to .fit files</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>write_weights</I><A NAME="19777"></A> = </FONT><FONT SIZE="-1"><TT>&lt;logical:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>False</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">whether to write the whole NxM weight matrix for ALIGN*</FONT></TD>
</TR>
<TR><TD ALIGN="LEFT"><FONT SIZE="-1"><I>write_whole_pdb</I><A NAME="19789"></A> = </FONT><FONT SIZE="-1"><TT>&lt;logical:1&gt;</TT> </FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1"><TT>True</TT></FONT></TD>
<TD ALIGN="LEFT"><FONT SIZE="-1">whether to write out all lines in the input PDB file</FONT></TD>
</TR>
</TABLE>

<P>
<DL>
<DT><STRONG>Output:</STRONG></DT>
<DD><I>error_status</I><A NAME="19799"></A> = <TT>&lt;integer:1&gt;</TT>
                             
</DD>
</DL>

<P>
<DL>
<DT><STRONG>Description:</STRONG></DT>
<DD>This command is a general dynamic programming based alignment
procedure for aligning sequences, structures or a combination of the two. It
is loosely based on the program COMPARER [<A
 HREF="node151.html#SalBlu90">&#352;ali &amp; Blundell, 1990</A>]. SALIGN can be used
to generate multiple protein structures/sequences alignments or to align two
blocks of sequences/structures that are in memory.

<P>
Broadly classifying, three different types of protein alignments categories
are tackled by this command:

<OL>
<LI>Multiple structure alignments
</LI>
<LI>Aligning a structure block to a sequence block
</LI>
<LI>Multiple and pair-wise protein sequence alignment
</LI>
</OL>

<P>
The command incorporates the functionality of the old M<SMALL>ODELLER</SMALL> commands -
(<B>alignment.align()</B><A NAME="19976"></A>, <B>alignment.align2d()</B><A NAME="19979"></A>, <B>alignment.malign()</B><A NAME="19982"></A>,
<B>alignment.align3d()</B><A NAME="19985"></A>, and <B>alignment.malign3d()</B><A NAME="19988"></A>). Some of the examples
below illustrate the equivalent script files to replace the old alignment
commands with <B>alignment.salign()</B><A NAME="19991"></A>.

<P>
In addition to these, this command has several new alignment features
including profile-profile sequence alignments and a dendrogram based multiple
sequence/structure alignment.

<P>
All pair-wise alignments make use of local or global dynamic programming.
A switch from one to another can be effected by setting <I>local_alignment</I><A NAME="19994"></A>
to <TT>True</TT> or <TT>False</TT>. The dynamic programming can be carried out using
affine gap penalties (as previously used in <B>alignment.align()</B><A NAME="19999"></A>) or an
environment dependent gap penalty function (as used in <B>alignment.align2d()</B><A NAME="20002"></A>).
The choice of gap penalty types can be regulated by switching the variable
<I>gap_function</I><A NAME="20005"></A> on or off. All arguments that associated to the
<B>alignment.align()</B><A NAME="20008"></A> and <B>alignment.align2d()</B><A NAME="20011"></A> commands apply.

<P>
</DD>
</DL>
<H3><A NAME="SECTION007422100000000000000">
Multiple alignments of protein structures (and sequences)</A>
</H3>

<P>
The description below is from the perspective of a multiple structure
alignment. Much of it holds true for aligning sequence multiply except that
the appropriate feature weights (see below for a detailed discussion on
feature weights) must be utilized.

<P>
When <I>alignment_type</I><A NAME="20014"></A> is set to <TT>tree</TT> this command performs a `tree'
based multiple alignment. The `tree' alignment algorithm consists of a
calculation of a dendrogram of the <IMG
 WIDTH="14" HEIGHT="13" ALIGN="BOTTOM" BORDER="0"
 SRC="img74.png"
 ALT="$n$"> proteins in memory, followed by a
progression of <IMG
 WIDTH="41" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img103.png"
 ALT="$n-1$"> pairwise alignments of the growing sub-alignments.
A sub-alignment is an alignment of <IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img104.png"
 ALT="$&lt;n$"> proteins. The pairwise alignment of
two sub-alignments is achieved by the <TT>'align_lingp'</TT> or the <TT>'align_vargp'</TT>
routines depending on whether <I>gap_function</I><A NAME="20020"></A> is set to <TT>False</TT> or <TT>True</TT>
(arguments to the <B>alignment.align()</B><A NAME="20025"></A> and <B>alignment.align2d()</B><A NAME="20028"></A> commands
apply).

<P>
The alignment of proteins within a sub-alignment does not change when the
sub-alignment is aligned with another protein or sub-alignment. The pairwise
alignment of sub-alignments is guided by the dendrogram. First, the most
similar pair of proteins are aligned. Second, the next most similar pair of
proteins are aligned, or the third protein is aligned with the sub-alignment
of the first two, as indicated by the dendrogram. This greedy, progressive
procedure requires <IMG
 WIDTH="41" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img103.png"
 ALT="$n-1$"> steps to align all <IMG
 WIDTH="14" HEIGHT="13" ALIGN="BOTTOM" BORDER="0"
 SRC="img74.png"
 ALT="$n$"> proteins, and each step
requires a pairwise alignment of two sub-alignments.

<P>
If <I>alignment_type</I><A NAME="20031"></A> is set to <TT>progressive</TT>, the multiple alignment
follows only the last part of the `tree' alignment where, in <IMG
 WIDTH="41" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img103.png"
 ALT="$n-1$"> alignments,
all the structures/sequences are successively aligned to the first one on the
list.

<P>
If a profile-profile alignment is to be performed, given two multiple sequence
alignments, <I>alignment_type</I><A NAME="20035"></A> should be set to <TT>pairwise</TT>.

<P>
Two multiple sequences alignments of proteins can be aligned using the
profile-profile option of this command. The two multiple alignments need to be
concatenated into one multiple alignment file. The end of the multiple
alignment of the first protein sequences is demarcated by setting
<I>align_block</I><A NAME="20039"></A> to the numerical value of the number of sequences in the
first multiple alignment.

<P>
<I>align_what</I><A NAME="20042"></A> is set to 'PROFILE' to indicate that profiles will be matched.
Since this is done for a pair of profiles, <I>alignment_type</I><A NAME="20045"></A> is set to
<TT>PAIRWISE</TT>. Profile comparisons can be done by comparing vectors of the
variation of the 20 amino acids per position(PSSM) or by comparing the
residue-residue substitution matrices implied at each alignment position of
the multiple sequence alignment. This is effected by setting
<I>comparison_type</I><A NAME="20049"></A> to <TT>PSSM</TT> or <TT>MAT</TT>.

<P>
If the comparison type is <TT>PSSM</TT> the correlation coefficient of the position
vectors is computed. If the comparison type is <TT>MAT</TT>, the user can choose to
take the maximum of residue-residue substitution at the aligned positions, or
the average of the correlation coefficient of the two matrices
(set <I>matrix_comparison</I><A NAME="20056"></A> to <TT>MAX</TT>, <TT>AVE</TT> or <TT>CC</TT>, respectively).
In the PSSM comparisons, the correlation coefficient can be multiplied by
the residue-residue substitution value by setting <I>substitution</I><A NAME="20062"></A> = <TT>True</TT>.
The smoothing of data with priors is regulated by setting
<I>smooth_prof_weight</I><A NAME="20066"></A> appropriately. When there are only the individual
sequences in the multiple sequence alignment, the alignment reduces to a
simple dynamic programming alignment (local or global) between the two
sequences if <I>substitution</I><A NAME="20069"></A> = <TT>True</TT>.

<P>
The <B>alignment.salign()</B><A NAME="20073"></A> command uses position-position dissimilarity
scores (except when <I>similarity_flag</I><A NAME="20076"></A> is switched on), as opposed to
similarity scores. This convention applies to all the features, including the
residue-residue similarities read from the <I>rr_file</I><A NAME="20079"></A>; however, if a residue
type - residue type similarity matrix is read in, it is automatically
converted into the distance matrix by <!-- MATH
 $D = \max_{i,j} S_{i,j}  - S$
 -->
<IMG
 WIDTH="141" HEIGHT="30" ALIGN="MIDDLE" BORDER="0"
 SRC="img105.png"
 ALT="$D = \max_{i,j} S_{i,j} - S$">.
In addition, it is also scaled linearly such that the residue-residue
dissimilarity scores range from 0 to 1 (to facilitate weighting this feature
with other features).

<P>
For each pairwise alignment, the weight matrix <IMG
 WIDTH="21" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img106.png"
 ALT="$W$"> has dimensions <IMG
 WIDTH="19" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img25.png"
 ALT="$N$"> and <IMG
 WIDTH="21" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img107.png"
 ALT="$M$">
that correspond to the lengths of the sub-alignments to be aligned based on
the weight matrix <IMG
 WIDTH="21" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img106.png"
 ALT="$W$">. The dissimilarity score for aligning position <IMG
 WIDTH="10" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img36.png"
 ALT="$i$"> with
position <IMG
 WIDTH="11" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img70.png"
 ALT="$j$"> is calculated as
<!-- MATH
 $W_{i,j} = \sum_f [ \frac{\omega_f}{\sum_f \omega_f} W^f_{i,j} ] + g_{i,j}$
 -->
<IMG
 WIDTH="206" HEIGHT="39" ALIGN="MIDDLE" BORDER="0"
 SRC="img108.png"
 ALT="$W_{i,j} = \sum_f [ \frac{\omega_f}{\sum_f \omega_f} W^f_{i,j} ] + g_{i,j}$">,
where the sum runs over all selected features <IMG
 WIDTH="14" HEIGHT="30" ALIGN="MIDDLE" BORDER="0"
 SRC="img109.png"
 ALT="$f$">, and <IMG
 WIDTH="12" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img110.png"
 ALT="$g$"> is a function that
may be used to correct the <IMG
 WIDTH="34" HEIGHT="30" ALIGN="MIDDLE" BORDER="0"
 SRC="img111.png"
 ALT="$W_{i,j}$"> score for the presence of gaps within the
sub-alignments (see below). A feature <IMG
 WIDTH="14" HEIGHT="30" ALIGN="MIDDLE" BORDER="0"
 SRC="img109.png"
 ALT="$f$"> is selected when its weight
<IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img112.png"
 ALT="$\omega_f$"> (specified in <I>feature_weights</I><A NAME="20082"></A>) is non-zero. The matrices
<IMG
 WIDTH="30" HEIGHT="17" ALIGN="BOTTOM" BORDER="0"
 SRC="img113.png"
 ALT="$W^f$"> are normalized to have the mean of 0 and standard deviation of 1 when
<I>normalize_pp_scores</I><A NAME="20085"></A> is <TT>True</TT>, but it is recommended not to use this
option for now (<EM>i.e.</EM>, use <I>feature_weights</I><A NAME="20090"></A> to scale the contributions of
the different features to the final <IMG
 WIDTH="21" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img106.png"
 ALT="$W$">). The weights of 1 will weigh the
different features approximately evenly (the residue-residue dissimilarities
of feature 1 are scaled to a range from 0 to 1, the position differences of
feature 2 are in angstroms, the fractional solvent accessibility scores of
feature 3 and the secondary structure scores of feature 4 range from 0 to 2,
and the DRMS difference of feature 5 is expressed in angstroms).

<P>
Feature 1 is the residue type. <IMG
 WIDTH="34" HEIGHT="34" ALIGN="MIDDLE" BORDER="0"
 SRC="img114.png"
 ALT="$W^1_{i,j}$"> is obtained from the residue
type - residue type dissimilarity matrix, specified in the file <I>rr_file</I><A NAME="20093"></A>.
<IMG
 WIDTH="34" HEIGHT="34" ALIGN="MIDDLE" BORDER="0"
 SRC="img114.png"
 ALT="$W^1_{i,j}$"> dissimiarity score for positions <IMG
 WIDTH="10" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img36.png"
 ALT="$i$"> and <IMG
 WIDTH="11" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img70.png"
 ALT="$j$"> in the two compared
sub-alignments is the average dissimilarity score for a comparison of all
residues in one sub-alignment with all residues in the other sub-alignment
(note that gaps are ignored here). Should only feature weight 1 be non-zero,
the user has an option of considering residue-residue similarity scores
instead of distances scores by setting <I>similarity_flag</I><A NAME="20096"></A> to <TT>True</TT>.

<P>
Feature 2 is the inter-molecular distance for a pair of residues (unless
<I>align3d_trf</I><A NAME="20100"></A> is <TT>True</TT>, see <B>alignment.align3d()</B><A NAME="20104"></A>). Only one atom per
residue is of course selected, as specified by <I>fit_atoms</I><A NAME="20107"></A> (<EM>e.g.</EM>, <IMG
 WIDTH="24" HEIGHT="30" ALIGN="MIDDLE" BORDER="0"
 SRC="img2.png"
 ALT="${C}_\alpha$">,
although we should also allow for <IMG
 WIDTH="24" HEIGHT="30" ALIGN="MIDDLE" BORDER="0"
 SRC="img115.png"
 ALT="${C}_\beta$"> in the future, which requires an
intervention for Gly). This `position' feature is complicated because it
depends on the relative orientation of the structures corresponding to the two
compared alignments. <IMG
 WIDTH="34" HEIGHT="34" ALIGN="MIDDLE" BORDER="0"
 SRC="img116.png"
 ALT="$W^2_{i,j}$"> is the Euclidean distance between the
compared positions <IMG
 WIDTH="10" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img36.png"
 ALT="$i$"> and <IMG
 WIDTH="11" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img70.png"
 ALT="$j$"> in the two compared sub-alignments that are
already optimally aligned and superposed based on their coordinates alone.
This optimal alignment is obtained by an iterative procedure as follows
(the same as in <B>alignment.align3d()</B><A NAME="20113"></A>). The average structures for both
sub-alignments are calculated for all sub-alignment positions with at least
one defined selected atom. This calculation is straightforward because the
structures within the two sub-alignments are already superposed with each
other (see below). Then, the distance matrix for dynamic programming by
<TT>'align_lingp'</TT> is calculated as the matrix of Euclidean distances between
the two averages. The dynamic programming results into a new alignment,
dependent also on the gap initiation and extension penalties
<I>gap_penalties_3d</I><A NAME="20117"></A> (a reasonable setting is 0 3).
<I>gap_penalties_3d</I><A NAME="20120"></A>[1] is a gap creation penalty (usually 0), and
<I>gap_penalties_3d</I><A NAME="20123"></A>[2] is a gap extension penalty, say 3. When the gap
initiation penalty is 0, pairs of positions are identified as equivalent
when they have their selected atoms at most 2 times <I>gap_penalties_3d</I><A NAME="20126"></A>[2]
angstroms apart in the current superposition, as described for the
<B>alignment.align3d()</B><A NAME="20129"></A> command. The new alignment is then used to generate
the new superposition of the two averages, and the iteration of the distance
matrix calculation, alignment and superposition is repeated until there are
no changes in the number of equivalent positions and in the rotation matrix
relating the two averages.

<P>
The values of both <I>improve_alignment</I><A NAME="20132"></A> and <I>fit</I><A NAME="20135"></A> are used in the
calculation of the position feature. That is, the initial alignment and
the orientation of the coordinates can be selected not to change at all
during the calculation of the inter-molecular distance matrix.

<P>
When the calculation of the inter-molecular distance matrix is finished, all
the structures in the second sub-alignment are rotated and translated
following the optimal rotation and translation of the second average on the
first average. These superpositions prepare the individual structures for the
next of the <IMG
 WIDTH="41" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img103.png"
 ALT="$n-1$"> stages of the progressive multiple alignment, and also
orient all the structures for writing out to <TT>'alignment[i].code_fit.pdb'</TT>
atom files if <I>write_fit</I><A NAME="20139"></A> = <TT>True</TT>. If <I>fit_pdbnam</I><A NAME="20143"></A> = <TT>False</TT>, the
PDB filenames in the output alignment file will not have the <TT>'_fit.pdb'</TT>
extensions. Thus, feature 2 needs to be selected by
<I>feature_weight</I><A NAME="20148"></A>[2]&nbsp;<IMG
 WIDTH="29" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img117.png"
 ALT="$&gt; 0$"> if you wish to write out the structures
superposed according to the tree-following procedure; also, <I>fit_on_first</I><A NAME="20151"></A>
must be <TT>False</TT>, otherwise the structures are written out superposed on the
first structure according to the final alignment (see also below).

<P>
The alignment produced within the routine that calculates <IMG
 WIDTH="28" HEIGHT="16" ALIGN="BOTTOM" BORDER="0"
 SRC="img118.png"
 ALT="$W^2$"> does not
generally correspond to the alignment calculated based on <IMG
 WIDTH="21" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img106.png"
 ALT="$W$">. Therefore, the
multiply superposed structures are not necessarily superposed based on the
final multiple alignment produced by <B>alignment.salign()</B><A NAME="20155"></A>. If you wish such
a superposition, you can use <B>alignment.malign3d()</B><A NAME="20158"></A> with <I>fit</I><A NAME="20161"></A> = <TT>False</TT>
and <I>write_fit</I><A NAME="20165"></A> = <TT>True</TT> (the meaning of <I>fit</I><A NAME="20169"></A> is different between
<B>alignment.salign()</B><A NAME="20172"></A> and <B>alignment.malign3d()</B><A NAME="20175"></A>).

<P>
Unless the position feature is selected, the initial alignment does not matter.
If the position feature is selected, a good starting alignment is a multiple
sequence alignment, obtained either by <B>alignment.malign()</B><A NAME="20178"></A> or by
<B>alignment.salign()</B><A NAME="20181"></A> used without the position feature (the initial
alignment can also be prepared using the position feature). If the position
feature is used, each pair of structures needs to have at least 3 aligned
residues at all points during the alignment.

<P>
There are several possibilities as to the final orientation of the input
coordinates. If <I>fit_on_first</I><A NAME="20184"></A> is <TT>True</TT>, all the coordinate sets are
superposed on the first structure, using the final multi-feature multiple
alignment. If <I>fit_on_first</I><A NAME="20188"></A> is <TT>False</TT>, and position feature was used,
and <I>fit</I><A NAME="20192"></A> was <TT>True</TT>, the coordinates will be superposed in the
progressive manner guided by the tree, by the routine that calculates the
inter-molecular distance matrices; this superposition is based only on the
positions of the selected atoms (feature 2), not on other features such as
residue type, secondary, structure, <EM>etc</EM>. If <I>improve_alignment</I><A NAME="20197"></A> is
<TT>False</TT>, it does not make much sense to have <I>fit</I><A NAME="20201"></A> = <TT>True</TT>
(use <I>fit_on_first</I><A NAME="20205"></A> = <TT>True</TT>).

<P>
Feature 3 is the fractional sidechain accessibility. The pair-wise
residue-residue dissimilarity is calculated by classifying residues into the
buried (<IMG
 WIDTH="37" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img119.png"
 ALT="$&lt;15$">%), semi-exposed, and exposed classes (<IMG
 WIDTH="37" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img120.png"
 ALT="$&gt;30$">%). The
dissimilarity is 0 for equal classes, 1 for neighboring classes and 2 for
the buried-exposed match.  The position.position dissimilarity is the
average residue-residue dissimilarity for comparing all residues from
one group to all residues in the other group (gaps are ignored).

<P>
Feature 4 is the secondary structure type, distinguishing between helix,
strand, and other. The pair-wise residue-residue dissimilarity is 0 for
equal classes, 1 for `helix' or `strand' matched to `other', and 2 for
`helix' matched to 'strand'. The position.position dissimilarity is the
average residue-residue dissimilarity for comparing all residues from
one group to all residues in the other group (gaps are ignored).

<P>
Feature 5 is the local conformation. A pair-wise residue-residue score
is DRMSD between the selected atoms (<I>fit_atoms</I><A NAME="20209"></A>) from the segments of
5 residues centered on the two matched residues. The position.position
dissimilarity is the average residue-residue dissimilarity for comparing
all residues from one group to all residues in the other group (gaps
are ignored).

<P>
Feature 6 is a user specified feature for which a external matrix (in 
M<SMALL>ODELLER</SMALL> matrix format; see the substitution matrices in the <TT>modlib</TT>
directory for examples) has to be specified using <I>input_weights_file</I><A NAME="20213"></A>.
The user can input either a similarity matrix (<I>weights_type</I><A NAME="20216"></A> = <TT>SIMILAR</TT>)
or a distance matrix (<I>weights_type</I><A NAME="20220"></A> = <TT>DISTANCE</TT>).

<P>
The gap correction function is
<!-- MATH
 $g_{i,j} = \frac{n_{rg}}{(n_1 n_2)} r + \frac{n_{gg}}{(n_1 n_2)} g$
 -->
<IMG
 WIDTH="167" HEIGHT="34" ALIGN="MIDDLE" BORDER="0"
 SRC="img121.png"
 ALT="$g_{i,j} = \frac{n_{rg}}{(n_1 n_2)} r + \frac{n_{gg}}{(n_1 n_2)} g$">,
where <IMG
 WIDTH="21" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img122.png"
 ALT="$n_1$"> and <IMG
 WIDTH="21" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img123.png"
 ALT="$n_2$"> are the number of proteins in the two sub-alignments,
<IMG
 WIDTH="27" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img124.png"
 ALT="$n_{rg}$"> is the number of gap-residue pairs, and <IMG
 WIDTH="28" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img125.png"
 ALT="$n_{gg}$"> is the number of
gap-gap pairs when comparing protein positions from one sub-alignment with
protein position from the other sub-alignment, <IMG
 WIDTH="12" HEIGHT="13" ALIGN="BOTTOM" BORDER="0"
 SRC="img126.png"
 ALT="$r$"> is <I>gap_residue_score</I><A NAME="20224"></A>
and <IMG
 WIDTH="12" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img110.png"
 ALT="$g$"> is <I>gap_gap_score</I><A NAME="20227"></A>. The smaller (even negative) is
<I>gap_gap_score</I><A NAME="20230"></A>, and the larger is <I>gap_residue_score</I><A NAME="20233"></A>, the more will
the gaps be aligned with gaps.

<P>
Argument <I>output</I><A NAME="20236"></A> can contain the following values:

<UL>
<LI><TT>'ALIGNMENT'</TT>: the alignments in the first <IMG
 WIDTH="41" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img127.png"
 ALT="$n-2$"> stages of the pairwise
alignment of sub-alignments are written out.

<P>
</LI>
<LI><TT>'QUALITY'</TT>: the final alignment is used to obtain pairwise least-squares
superpositions and the corresponding average and minimal numbers of pairs of
aligned residues that are within <I>rms_cutoffs</I><A NAME="20241"></A>[1] &#197; in all pairs of
aligned structures. These numbers can be used as absolute quality measures
for the final multiple alignment. This option requires the coordinate files
for the aligned proteins.
</LI>
</UL>

<P>
The user can choose to have certain alignment positions "fixed" by offsetting
the appropriate matrix entries. This is achieved by setting <I>fix_offset</I><A NAME="20244"></A>
to <TT>True</TT>. Below the given pair of sequence blocks in the input alignment
file, a new pseudo sequence is introduced with the align code
<TT>._fix_pos</TT>. The residues of this pseudo sequence are either blank
spaces or integer values from 0 (blanks are assigned 0 by default) through 4.
The placement of the non-zero integers is done so that they are below the
positions that are to be offset, in an effort to ensure that the alignment
implied by that position is restrained. The extent to which this alignment
position is restrained depends on the offset that the numerical value (0-4)
corresponds to. This can also be specified by the user by specifying the
matrix offset values with <I>fix_offsets</I><A NAME="20248"></A>.

<P>
Suboptimal alignments. The weight matrix can be offset at random, many times
over, to generate several `sub-optimal' alignments. The number of sub-optimal
alignments to be output can be specified with <I>n_subopt</I><A NAME="20251"></A>. Though the matrix
positions at which these offsets cannot be controlled, the user can choose
by how much the matrix will be offset (<I>subopt_offset</I><A NAME="20254"></A>).

<P>
In addition, if you enable verbose logging with <B>log.verbose()</B><A NAME="20257"></A>, there will
be more output in the <TT>'log'</TT> file, such as the dendrogram. The dendrogram can
also be written out in a separate file by specifying the file name to
<I>dendrogram_file</I><A NAME="20261"></A>.

<P>
If <I>write_fit</I><A NAME="20264"></A> is <TT>True</TT>, the fitted atom files are written out in their
fitted orientations. For this and other options below, also read the text
above.

<P>
If <I>write_weights</I><A NAME="20268"></A> is <TT>True</TT>, the dynamic programming weight matrix is
written out into the file specified by <I>output_weights_file</I><A NAME="20272"></A>.

<P>
If <I>current_directory</I><A NAME="20275"></A> is <TT>True</TT>, the output <TT>_pdb.fit</TT> files will
be written to the current directory. Otherwise, the output will be in the
directory with the original files.

<P>
If <I>write_whole_pdb</I><A NAME="20279"></A> is <TT>True</TT>, the whole PDB files are written out;
otherwise only the parts corresponding to the aligned sequences are output.

<P>
If <I>fit</I><A NAME="20283"></A> is <TT>False</TT>, the initial superposition is not changed. This is
useful when all the structures have to be compared with a given alignment
as is, without changing their relative orientation.

<P>
If <I>fit_on_first</I><A NAME="20287"></A> is <TT>True</TT>, the structures are fit to the first
structure according to the final alignment before they are written out.

<P>
If <I>improve_alignment</I><A NAME="20291"></A> is <TT>False</TT>, the initial alignment is not changed,
though the structures may still be superimposed if <I>fit</I><A NAME="20295"></A> = <TT>True</TT>. This
is useful when all the structures have to be superimposed with the initial
alignment.

<P>

<DL>
<DT><STRONG>Example:</STRONG></DT>
<DD> <BR>
<BR>
<PRE>[blank]
</PRE>
</DD>
</DL>  <BR>
<P>
<HR>
<!--Navigation Panel-->
<A NAME="tex2html1803"
  HREF="node85.html">
<IMG WIDTH="37" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="next" SRC="next.png"></A> 
<A NAME="tex2html1797"
  HREF="node62.html">
<IMG WIDTH="26" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="up" SRC="up.png"></A> 
<A NAME="tex2html1791"
  HREF="node83.html">
<IMG WIDTH="63" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="previous" SRC="prev.png"></A> 
<A NAME="tex2html1799"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="contents" SRC="contents.png"></A> 
<A NAME="tex2html1801"
  HREF="node152.html">
<IMG WIDTH="43" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="index" SRC="index.png"></A> 
<BR>
<B> Next:</B> <A NAME="tex2html1804"
  HREF="node85.html">alignment.to_profile()   convert</A>
<B> Up:</B> <A NAME="tex2html1798"
  HREF="node62.html">The alignment class: comparison</A>
<B> Previous:</B> <A NAME="tex2html1792"
  HREF="node83.html">alignment.malign3d()   align</A>
 &nbsp; <B>  <A NAME="tex2html1800"
  HREF="node1.html">Contents</A></B> 
 &nbsp; <B>  <A NAME="tex2html1802"
  HREF="node152.html">Index</A></B> 
<!--End of Navigation Panel-->
<ADDRESS>
Ben Webb
2005-01-26
</ADDRESS>
</BODY>
</HTML>
